### ISSUE TITLE
Incorrect code generation for fragments of the same type

### ISSUE THREAD

**Issue by axelhzf** [dotansimha/graphql-code-generator#808]
**Describe the bug**

**To Reproduce**
1. schema

```gql
type User {
  id: String!
  displayName: String!
  lastLoginAt: String!
}

type Query {
  users: [User!]!
}
```
2. document

```gql
  fragment A on User {
    id
    displayName
  }  
  
  fragment B on User {
    id
    lastLoginAt
  }
  
  query Test { 
    users {
      ...A
      ...B
    }
  }
```

The generated code is

```ts
User = A.Fragment | B.Fragment
```

**Expected behavior**

I think the correct type in that case is 

```ts
User = A.Fragment & B.Fragment
```

**Schema/Documents**
If possible and relevant, please provide your GraphQL schema and GraphQL documents.

**Environment:**
 - OS: osx
 - Codegen: v0.13.0
 - Node: v8.11.3

**Additional context**

Maybe the problem is related with this change https://github.com/dotansimha/graphql-code-generator/pull/737

**Comment by mvestergaard** [comment]
We're also impacted by this. Had to downgrade to 0.12.6

**Comment by ardatan** [comment]
I think `or` operator is decent one there, because a result cannot have properties of both AFragment and BFragment, am I missing something?

**Comment by mvestergaard** [comment]
It can, look at the query from the example above:
```gql
query Test { 
    users {
      ...A
      ...B
    }
  }
```

Both fragments `A` and `B` is used, so it's not `A | B`, it's `A & B`

**Comment by ardatan** [comment]
Oh my mistake, you're right!

**Comment by kamilkisiela** [comment]
#894

**Comment by dotansimha** [comment]
Fixed in 0.14.2 🎉
